functions.php

```diff
@@ -1319,3 +1340,271 @@ function wp_autoupdates_themes_bulk_actions_handle( $redirect_to, $doaction, $it
+			$enabled_count         = count( $wp_autoupdate_plugins );
+			$plugin_name           = $all_plugins[ $asset ]['Name'];
+
+			$plugin_url = add_query_arg(
+				array(
+					'action'   => 'autoupdate',
+					'plugin'   => $asset,
+					'_wpnonce' => wp_create_nonce( 'autoupdate-plugin_' . $asset ),
+				),
+				'plugins.php'
+			);
+
+			$aria_label = esc_attr(
+				sprintf(
+					/* translators: Plugin name. */
+					_x( 'Enable automatic updates for %s', 'plugin', 'wp-autoupdates' ),
+					esc_html( $plugin_name )
+				)
+			);
+			$return_html  = '<p>';
+			$return_html .= '<span class="plugin-autoupdate-enabled">' . esc_html__( 'Auto-updates enabled', 'wp-autoupdates' ) . '</span>';
+			$return_html .= '<br />';
+			$return_html .= sprintf(
+				'<a href="%s" class="plugin-autoupdate-disable" aria-label="%s">%s</a>',
+				esc_url_raw( $plugin_url ),
+				$aria_label,
+				esc_html__( 'Disable auto-updates', 'wp-autoupdates' )
+			);
+			$return_html .= '</p>';
+
+			wp_send_json_success(
+				array(
+					'enabled_count'  => '(' . $enabled_count . ')',
+					'disabled_count' => '(' . absint( count( $all_plugins ) - $enabled_count ) . ')',
+					'return_html'    => $return_html,
+					'type'           => 'plugin',
+				)
+			);
+		}
+	} elseif ( 'theme' === $type ) {
+		$wp_autoupdate_themes = get_site_option( 'wp_auto_update_themes', array() );
+		if ( is_array( $wp_autoupdate_themes ) ) {
+			$wp_autoupdate_themes[] = $asset;
+			array_unique( $wp_autoupdate_themes );
+			update_site_option( 'wp_auto_update_themes', $wp_autoupdate_themes );
+
+			$all_themes           = wp_get_themes();
+			$wp_autoupdate_themes = array_intersect( $wp_autoupdate_themes, array_keys( $all_themes ) );
+			$enabled_count        = count( $wp_autoupdate_themes );
+			$theme_name           = $all_themes[ $asset ]->get( 'Name' );
+
+			$theme_url = add_query_arg(
+				array(
+					'action'   => 'autoupdate',
+					'theme'    => $asset,
+					'_wpnonce' => wp_create_nonce( 'autoupdate-theme_' . $asset ),
+				),
+				'themes.php'
+			);
+
+			$aria_label = esc_attr(
+				sprintf(
+					/* translators: Plugin name. */
+					_x( 'Enable automatic updates for %s', 'theme', 'wp-autoupdates' ),
+					esc_html( $theme_name )
+				)
+			);
+			$return_html  = '<p>';
+			$return_html .= '<span class="theme-autoupdate-enabled">' . esc_html__( 'Auto-updates enabled', 'wp-autoupdates' ) . '</span>';
+			$return_html .= '<br />';
+			$return_html .= sprintf(
+				'<a href="%s" class="theme-autoupdate-disable" aria-label="%s">%s</a>',
+				esc_url_raw( $theme_url ),
+				$aria_label,
+				esc_html__( 'Disable auto-updates', 'wp-autoupdates' )
+			);
+			$return_html .= '</p>';
+
+			wp_send_json_success(
+				array(
+					'enabled_count'  => '(' . $enabled_count . ')',
+					'disabled_count' => '(' . absint( count( $all_themes ) - $enabled_count ) . ')',
+					'return_html'    => $return_html,
```

Comment:
Same comment here as above about thinking about late escaping.

Reply:
@whyisjake is `wp_kses_post` sufficient for late escaping of the HTML, or do you want to be stricter than that?

Reply:
See https://github.com/WordPress/wp-autoupdates/pull/61/commits/af2480beabeaaa81853b0ef598b48948d54769e9 please.

Reply:
WFM.